lib/trino-record-decoder/src/main/java/io/trino/decoder/protobuf/ProtobufColumnDecoder.java

```diff
@@ -118,18 +136,50 @@ private static boolean isSupportedPrimitive(Type type)
 
     public FieldValueProvider decodeField(DynamicMessage dynamicMessage)
     {
-        return new ProtobufValueProvider(locateField(dynamicMessage, columnMapping), columnType, columnName);
+        return new ProtobufValueProvider(locateField(dynamicMessage, columnMapping, oneOfs), columnType, columnName);
     }
 
     @Nullable
-    private static Object locateField(DynamicMessage message, String columnMapping)
+    private static Object locateField(DynamicMessage message, String columnMapping, Optional<List<String>> oneofColumns)
     {
         Object value = message;
         Optional<Descriptor> valueDescriptor = Optional.of(message.getDescriptorForType());
         for (String pathElement : Splitter.on('/').omitEmptyStrings().split(columnMapping)) {
             if (valueDescriptor.filter(descriptor -> descriptor.findFieldByName(pathElement) != null).isEmpty()) {
-                return null;
+                if (oneofColumns.isPresent()) {
+                    DynamicSliceOutput output = new DynamicSliceOutput(40);
```

Comment:
can we come up with better estimates, e.g. based on field count * some avg length?

Reply:
I made a change to calculate the size based on the string values of the field name, value, plus the extra characters. Looking at the data type test case, there wouldn't be any resizing.